### PR TITLE
Move Elim-specific code

### PR DESCRIPTION
**Pull request by ppedrot**
Move elim-specific code from Tacticals to Elim, and cleanup the Elim API.

### CODE REVIEW

File: tactics/elim.ml [hunk 1]
```diff
@@ -8,35 +8,148 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
+open CErrors
```
Comment (SkySkimmer) [inline tactics/elim.ml:11]: Could we keep CErrors closed?

Comment (ppedrot) [reply tactics/elim.ml:11]: OK, but what's the motivation?

Comment (SkySkimmer) [reply tactics/elim.ml:11]: I can't put it in clear terms but I prefer it, so if the cost is low it would be nice

Comment (ppedrot) [reply tactics/elim.ml:11]: Fine, I'll push an additional patch.

**Comment by ppedrot** [comment]
This is fairly trivial and needs an assignee.

**Review by herbelin — APPROVED** [review]
The move makes sense.

Probability is low that the removed API is used in plugins.

Typo in a commit message (ununed -> unused).

**Comment by herbelin** [comment]
> This is fairly trivial and needs an assignee.

Can do it. Will merge tomorrow.

**Comment by herbelin** [comment]
I took the initiative to seize the opportunity of merging to amend myself the typo in the commit message. I hope it is ok for you.

**Pull request merged**